integration_tests/src/main/python/parquet_write_test.py

```diff
@@ -544,6 +544,19 @@ def test_fallback_to_single_writer_from_concurrent_writer(spark_tmp_path, aqe_en
             {"spark.sql.adaptive.enabled": aqe_enabled},
         ))
 
+def test_write_exceed_max_file_counter_error(spark_tmp_path):
+    gen_list = [("c1", BooleanGen())]
+    data_path = spark_tmp_path + 'PARQUET_DATA'
+    assert_gpu_error(
```

Comment:
Why only check gpu here ? Shouldn't we check both cpu and gpu to see if they behave the same for this case ?